charts/fleet/templates/rbac_gitjob.yaml

```
@@ -43,6 +43,19 @@ rules:
       - 'events'
     verbs:
       - '*'
+  - apiGroups:
+      - ""
+    resources:
+      - serviceaccounts
+    verbs:
+      - "create"
+  - apiGroups:
+      - rbac.authorization.k8s.io
+    resources:
+      - roles
+      - rolebindings
+    verbs:
+      - "*"
```

Wouldn't this be enough?
```suggestion
      - "create"
```

---

I tried that, but it was not enough. 
I even tried adding verbs one by one. 
I get the following error, I guess because later it tries to grand RBAC permissions that are different to "create".

```bash
Reconciler error	{"controller": "gitrepo", "controllerGroup": "fleet.cattle.io", "controllerKind": "GitRepo", "GitRepo": {"name":"sample","namespace":"fleet-local"}, "namespace": "fleet-local", "name": "sample", "reconcileID": "2ad237ee-5d4a-49c9-9cb9-b38b1273ceb4", "error": "roles.rbac.authorization.k8s.io \"git-sample\" is forbidden: user \"system:serviceaccount:cattle-fleet-system:gitjob\" (groups=[\"system:serviceaccounts\" \"system:serviceaccounts:cattle-fleet-system\" \"system:authenticated\"]) is attempting to grant RBAC permissions not currently held:\n{APIGroups:[\"fleet.cattle.io\"], Resources:[\"bundles\"], Verbs:[\"get\" \"create\" \"update\" \"list\" \"delete\"]}\n{APIGroups:[\"fleet.cattle.io\"], Resources:[\"imagescans\"], Verbs:[\"get\" \"create\" \"update\" \"list\" \"delete\"]}"}
``` 

I even tried with 
* create
* delete
* deletecollection
* get
* list
* patch
* update
* watch 

and I got the same error.

This is still a mistery to me.
I ended up adding `*` because the original RBAC had exactly the same, more than happy to hold this PR to discuss this and understand why.

---

Maybe something like this then?
```
    verbs:
      -  'get'
      - 'create'
      - 'update'
      - 'list'
      - 'delete'
```

---

I tried that combination as well and I got the same error. :confused:

---

As discussed offline, I had to add `escalate` and `bind` to the `roles`.
It ended up like this:
```yaml
  - apiGroups:
      - ""
    resources:
      - serviceaccounts
    verbs:
      - "create"
  - apiGroups:
      - rbac.authorization.k8s.io
    resources:
      - roles
    verbs:
      - escalate
      - create
      - bind
  - apiGroups:
      - rbac.authorization.k8s.io
    resources:
      - rolebindings
    verbs:
      - create
```

---

Nice find :+1: Thanks!